### ISSUE TITLE
[iOS] RadioButton with custom Content cannot be selected with VoiceOver

### ISSUE THREAD

**Issue by rachelkang** [dotnet/maui#11583]
### Description

On iOS, a RadioButton with custom content cannot be selected when VoiceOver is activated. Double tapping should select the button (fill the radioButton selected circle).

This works as expected without VoiceOver and with simple RadioButtons + VoiceOver.

### Steps to Reproduce

1. Run .NET MAUI control gallery on iOS device
2. Navigate to RadioButton galleries > RadioButton ContentProperties page
3. Turn on VoiceOver
4. Tap "It's a button inside a button" button
5. Double tap to toggle. Notice button does not get selected

### Link to public reproduction project repository

https://github.com/dotnet/maui/

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 16.1

**Comment by rachelkang** [comment]
still an issue
